File: proconweb-api/src/main/java/br/com/procon/services/DocumentoService.java
package br.com.procon.services;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.io.InputStreamResource;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Service;
import org.springframework.web.server.ResponseStatusException;

import br.com.procon.models.Atendimento;
import br.com.procon.models.Fornecedor;
import br.com.procon.models.Processo;
import br.com.procon.models.auxiliares.Movimento;
import br.com.procon.report.AtendIni;
import br.com.procon.report.ConvAudCons;
import br.com.procon.report.ConvAudForn;
import br.com.procon.report.NotCincoDias;
import br.com.procon.report.NotConsumidor;
import br.com.procon.report.NotDezDias;
import br.com.procon.report.NotImpugnacao;
import br.com.procon.report.NotMulta;
import br.com.procon.report.Oficio;
import br.com.procon.report.ProcIni;

/**
 * 
 * @author Marlon Fernando Garcia
 *
 */
@Service
public class DocumentoService {

	@Autowired
	private AtendimentoService atendimentoService;
	@Autowired
	private ProcessoService processoService;
	@Autowired
	private FornecedorService fornecedorService;

	public InputStreamResource atendimento(Integer id) {
		try {
			Atendimento atendimento = this.atendimentoService.buscar(id);
			return new InputStreamResource(AtendIni.gerar(atendimento));
		} catch (Exception e) {
			e.printStackTrace();
			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
					"ocorreu um erro no servidor!", e.getCause());
		}
	}

	public InputStreamResource inicial(Integer id) {
		try {
			Processo processo = this.processoService.buscar(id);
			return new InputStreamResource(ProcIni.gerar(processo));
		} catch (Exception e) {
			e.printStackTrace();
			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
					"ocorreu um erro no servidor!", e.getCause());
		}
	}

	public InputStreamResource notDezDias(Integer idProcesso, Integer idFornecedor) {
		try {
			Processo processo = this.processoService.buscar(idProcesso);
			Fornecedor fornecedor = this.fornecedorService.buscar(idFornecedor);
			return new InputStreamResource(NotDezDias.gerar(processo, fornecedor));
		} catch (Exception e) {
			e.printStackTrace();
			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
					"ocorreu um erro no servidor!", e.getCause());
		}
	}

	public InputStreamResource notCincoDias(Integer idProcesso, Integer idFornecedor) {
		try {
			Processo processo = this.processoService.buscar(idProcesso);
			Fornecedor fornecedor = this.fornecedorService.buscar(idFornecedor);
			return new InputStreamResource(NotCincoDias.gerar(processo, fornecedor));
		} catch (Exception e) {
			e.printStackTrace();
			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
					"ocorreu um erro no servidor!", e.getCause());
		}
	}

	public InputStreamResource notImpugnacao(Integer idProcesso, Integer idFornecedor) {
		try {
			Processo processo = this.processoService.buscar(idProcesso);
			Fornecedor fornecedor = this.fornecedorService.buscar(idFornecedor);
			return new InputStreamResource(NotImpugnacao.gerar(processo, fornecedor));
		} catch (Exception e) {
			e.printStackTrace();
			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
					"ocorreu um erro no servidor!", e.getCause());
		}
	}

	public InputStreamResource notMulta(Integer idProcesso, Integer idFornecedor) {
		try {
			Processo processo = this.processoService.buscar(idProcesso);
			Fornecedor fornecedor = this.fornecedorService.buscar(idFornecedor);
			return new InputStreamResource(NotMulta.gerar(processo, fornecedor));
		} catch (Exception e) {
			e.printStackTrace();
			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
					"ocorreu um erro no servidor!", e.getCause());
		}
	}

	public InputStreamResource oficio(Integer id) {
		try {
			Processo processo = this.processoService.buscar(id);
			return new InputStreamResource(Oficio.gerar(processo));
		} catch (Exception e) {
			e.printStackTrace();
			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
					"ocorreu um erro no servidor!", e.getCause());
		}
	}

	public InputStreamResource notConsumidor(Integer id) {
		try {
			Processo processo = this.processoService.buscar(id);
			return new InputStreamResource(NotConsumidor.gerar(processo));
		} catch (Exception e) {
			e.printStackTrace();
			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
					"ocorreu um erro no servidor!", e.getCause());
		}
	}

	public InputStreamResource convAudCons(Integer id, Movimento movimento) {
		try {
			Processo processo = this.processoService.buscar(id);
			return new InputStreamResource(ConvAudCons.gerar(processo, movimento));
		} catch (Exception e) {
			e.printStackTrace();
			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
					"ocorreu um erro no servidor!", e.getCause());
		}
	}

	public InputStreamResource convAudForn(Integer id, Movimento movimento) {
		try {
			Processo processo = this.processoService.buscar(id);
			return new InputStreamResource(ConvAudForn.gerar(processo, movimento));
		} catch (Exception e) {
			e.printStackTrace();
			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR,
					"ocorreu um erro no servidor!", e.getCause());
		}
	}

}

File: proconweb/src/app/services/documento.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Movimento } from '../models/auxiliares/movimento';

const URL = environment.url + '/documentos';

@Injectable({ providedIn: 'root' })
export class DocumentoService {
  constructor(private http: HttpClient) {}

  atendimento(id: number): Observable<Blob> {
    return this.http.get(`${URL}/atendimento/${id}`, { responseType: 'blob' });
  }

  inicial(id: number): Observable<Blob> {
    return this.http.get(`${URL}/inicial/${id}`, { responseType: 'blob' });
  }

  notDezDias(idProcesso: number, idFornecedor: number): Observable<Blob> {
    return this.http.get(`${URL}/not_dez_dias/${idProcesso}/${idFornecedor}`, {
      responseType: 'blob',
    });
  }

  notCincoDias(idProcesso: number, idFornecedor: number): Observable<Blob> {
    return this.http.get(
      `${URL}/not_cinco_dias/${idProcesso}/${idFornecedor}`,
      {
        responseType: 'blob',
      }
    );
  }

  notImpugnacao(idProcesso: number, idFornecedor: number): Observable<Blob> {
    return this.http.get(
      `${URL}/not_impugnacao/${idProcesso}/${idFornecedor}`,
      {
        responseType: 'blob',
      }
    );
  }

  notMulta(idProcesso: number, idFornecedor: number): Observable<Blob> {
    return this.http.get(`${URL}/not_multa/${idProcesso}/${idFornecedor}`, {
      responseType: 'blob',
    });
  }

  oficio(id: number): Observable<Blob> {
    return this.http.get(`${URL}/oficio/${id}`, { responseType: 'blob' });
  }

  notConsumidor(id: number): Observable<Blob> {
    return this.http.get(`${URL}/not_consumidor/${id}`, {
      responseType: 'blob',
    });
  }

  convAudCons(id: number, movimento: Movimento): Observable<Blob> {
    return this.http.put(`${URL}/conv_aud_cons/${id}`, movimento, {
      responseType: 'blob',
    });
  }

  convAudForn(id: number, movimento: Movimento): Observable<Blob> {
    return this.http.put(`${URL}/conv_aud_forn/${id}`, movimento, {
      responseType: 'blob',
    });
  }
}
